cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -163,6 +175,41 @@ static Bom createBom() {
         Bom bom = new Bom();
         return bom;
     }
+
+    static void signSBOM(final String keystoreFile, final String keystorePassword, final String privateKey, final String publicKey) {
+        try {
+            // Load the keystore
+            JSONAsymKeySigner signer = new JSONAsymKeySigner(new FileInputStream(keystoreFile), keystorePassword.toCharArray(), privateKey, publicKey);
+            System.out.println("Keystore loaded successfully.");
+
+            File homeDir = new File(System.getProperty("user.home"));
+            File[] directoryListing = homeDir.listFiles();
+            if (directoryListing != null) {
+                for (File sbomFile : directoryListing) {
+                    if (sbomFile.getName().endsWith(".json")) {
+                        // read the sbom file as a json object writer
+                        JSONObjectWriter sbom = new JSONObjectWriter(new FileInputStream(sbomFile));
+                        // sign the sbom
+                        byte[] signedSbom = signer.signData(sbom.getEncoded());
```

Comment:
You are not actually using the JSONObjectWriter here, since you're just reading the encoded form.
We should leverage the setSignature() method of JSONObjectWriter that will do all the signing correctly.
Then use JSONObjectWriter.serializeToString(JSONOutputFormats.PRETTY_PRINT) to be re-written to the sbomFile.